src/streaming/text/TextTracks.js

```diff
@@ -472,9 +469,21 @@ function TextTracks() {
     }
 
     function setCurrentTrackIdx(idx) {
+        if (idx === currentTrackIdx) {
+            return;
+        }
         currentTrackIdx = idx;
-        let track = currentTrackIdx >= 0 && textTrackQueue[currentTrackIdx] ? videoModel.getTextTrack(textTrackQueue[currentTrackIdx].kind, textTrackQueue[currentTrackIdx].label, textTrackQueue[currentTrackIdx].lang) : null;
+        const track = getTrackByIdx(currentTrackIdx);
         setCueStyleOnTrack.call(this, track);
+
+        if (videoSizeCheckInterval) {
+            clearInterval(videoSizeCheckInterval);
+            videoSizeCheckInterval = null;
+        }
+
+        if (track && track.renderingType === 'html') {
```

Comment:
Hi @epiclabsDASH,

I think we have to check video size as soon as possible, not wait 500 ms. It should be better to call rigth now checkVideoSize and then start timer in order to avoid this  : 
![checkvideosize](https://user-images.githubusercontent.com/4638260/30802029-84c71088-a1e5-11e7-8264-306c56066f8f.png)

Do you agree?
Nico

Reply:
Fully agree!

Reply:
and change applied